### PR TITLE
Nuttx v3 - nuttx_next_cmake merged latest master

### PR DESCRIPTION
**Pull request by davids5**
This PR is a merge of the latest master (see below) via merges and cherry-picking in preparation for replacing master.

Current as of: 29550a4cee5618e3508f0d8dfa97bcbedece30da  

This PR will superceed https://github.com/PX4/Firmware/tree/nuttx_next_cmake 

**N.B. RAN OUT OF CODE SPACE** 
commits:

1.c7db20e removes LPE

esc35-v1 bootloader has been added. (the app - esc35-v1_default) is not tested - Do not use it!

The following are turned off on px4fmu-v2
new logger
drivers/lis3mdl
drivers/blinkm

# Testing nuttx_next

## Checking out the source

The upstream Nuttx GIT structure, and hence nuttx_v3, is different from Firmware/master 's Nuttx structure. 

When switching from master to nuttx_v3 use the following commands:
```
git submodule sync --recursive && git submodule update --init --recursive
make clean
```

When switching from nuttx_v3 to master use the following commands:
```
rm -fr NuttX
git submodule sync --recursive && git submodule update --init --recursive
make clean
```

### Compiler

```
arm-none-eabi-gcc (GNU Tools for ARM Embedded Processors) 5.4.1 20160609 (release) [ARM/embedded-5-branch revision 237715]
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

``` make px4fmu-v2_default upload```

## Building with upload

``` make px4fmu-v2_default upload```

## Building other targets

use ``` make list_config_targets```

to list the additional targets. N.B .Omit ```[nuttx_]``` from the build command.

### CODE REVIEW

**Comment by davids5** [comment]
Jenkins test this please

**Comment by kd0aij** [comment]
https://github.com/PX4/Firmware/issues/4634

**Comment by davids5** [comment]
Jenkins test this please

**Comment by davids5** [comment]
@LorenzMeier FYI current master on NuttX v1.17+

**Comment by LorenzMeier** [comment]
Jenkins ok to test.

**Comment by davids5** [comment]
Jenkins test this please

**Comment by santiago3dr** [comment]
Three seperate flights using nuttxV3
all stable, same copter, no issues to report

http://logs.uaventure.com/view/D6Nj53tLtGpbQUwnTPAKVf
http://logs.uaventure.com/view/x2nSdAbGwyUgkpjABvTjmA
http://logs.uaventure.com/view/izJzC6oEMTmhyyH8v5eu4M

**Comment by LorenzMeier** [comment]
Awesome, thanks!

File: CMakeLists.txt [hunk 1]
```diff
@@ -256,7 +256,7 @@ else()
 	#message(STATUS "catkin DISABLED")
 endif()
 
-find_package(PythonInterp 2.7 REQUIRED)
+find_package(PythonInterp REQUIRED)
```
Comment (zehortigoza) [inline CMakeLists.txt]: Why did you removed the python version?

Comment (davids5) [inline CMakeLists.txt]: That is from master
See https://github.com/PX4/Firmware/commit/7a698952fbe386d190a40cf3ba68fba18d1fbfa6

**Comment by LorenzMeier** [comment]
@davids5 I will break this up into a couple dozen commits over the weekend. Either I or we do it together.

**Comment by LorenzMeier** [comment]
Merged to master, yay!

**Pull request closed without merging**